crypto/merkletrie/node.go

```diff
@@ -152,24 +153,21 @@ func (n *node) add(cache *merkleTrieCache, d []byte, path []byte) (nodeID stored
 			}
 			pnode2.hash = append(path, d[:i]...)
 
-			pnode = pnode2
 			nodeID = nodeID2
 		}
 		return nodeID, nil
 	}
 
 	if n.childrenMask.Bit(d[0]) == false {
 		// no such child.
-		var childNode *node
-		var childNodeID storedNodeIdentifier
-		childNode, childNodeID = cache.allocateNewNode()
+		childNode, childNodeID := cache.allocateNewNode()
 		childNode.hash = d[1:]
 
 		pnode, nodeID = cache.allocateNewNode()
 		pnode.childrenMask = n.childrenMask
 		pnode.childrenMask.SetBit(d[0])
 
-		pnode.children = make([]childEntry, len(n.children)+1, len(n.children)+1)
+		pnode.children = make([]childEntry, len(n.children)+1)
```

Comment:
quick question: removing the capacity in `make` is because `pnode.children` size won't change after the initialization, right?

Reply:
`make(T, len, cap)` is the same as `make(T, len)`